org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenImpl.java

```diff
@@ -632,14 +602,6 @@ public Artifact resolve(String groupId, String artifactId, String version, Strin
     return resolve(artifact, remoteRepositories, monitor);
   }
 
-  private IMavenExecutionContext context() {
```

Comment:
IMO, it's better to make it an API here than to have a static accessor API in IMavenExecutionContext; because how to choose or create the executionContext depends on the IMaven instance (it uses `createExecutionContext()`).

Reply:
See comments above, if one wants to distinguish the cases one needs to use `canJoin()/join()` instead, because code using this currently explicitly is "thread context aware" and code that don't want to creates own ones in the first place.